enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,349 @@
+- Detect newly attached disks
+
+### Non-Goals
+
+- Inherently protect against confilict with provisioners that own local devices on the same nodes automatic detection is configured to run.
+
+### Risks and Mitigations
+- There is a risk that LSO detects a volume that contains data and we mitigate this risk by ensuring that the device:
+  - can be openened exclusively.
+  - is not read-only.
+  - is not removable.
+  - has no child partitions.
+  - has no FS signature.
+  - state (as outputted by `lsblk`) is not `suspended`
+- There is a risk that LSO will detect just attached device as empty, before kubelet formats it, we mitigate this risk 
+    - by re-checking the device after ~1 minute.
+- There is a risk that the path of the disk changes after reboot and disk can be re-detected as new, we mitigate this risk by
+    - Using disk id for symlink.
+- There is a risk that LSO will match all local disks that already have PVs, but these PVs are not bound / used yet.`
+    - We leave it up to the cluster admin for selecting the inclusion spec.
+- There is a risk that multiple `LocalVolumeGroup` and `LocalVolume` CRs will use the same device.
+    - We leave it up to the cluster admin to adjust the CR.
+
+## Proposal
+
+The `local-storage-operator` is already capable of consuming local disks from the nodes and provisioning PVs out of them,
+but the disk/device paths needs to explicitly specified in the `LocalVolume` CR.
+The idea is to introduce a new feature in LSO
+  - Auto provisioning of localDevices
+    - This will involve introduction of a new CR called `LocalVolumeGroup`.
+    - The pupose of this will be to auto discover and provision PVs on devices which match the inclusion filter present in the CR.
+    - This will involve a continous process of discovery of devices via the diskmaker daemons. Any discovered devices which matches the inclusion filter will be considered for provisioning of PVs.
+
+## Design Details for `Auto provisioning of local devices`
+API scheme for `LocalVolumeGroups`:
+
+```go
+type LocalVolumeGroupList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []LocalVolumeGroup `json:"items"`
+}
+
+type LocalVolumeGroup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              LocalVolumeGroupSpec   `json:"spec"`
+	Status            LocalVolumeGroupStatus `json:"status,omitempty"`
+}
+
+type LocalVolumeGroupSpec struct {
+	// Nodes on which the automatic detection policies must run.
+	// +optional
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	// StorageClassName to use for set of matched devices
+	StorageClassName string `json:"storageClassName"`
+	// MinDeviceCount is the minumum number of devices that needs to be detected per node.
+	// If the match devices are less than the minCount specified then no PVs will be created.
+	MinDeviceCount int `json:"minDeviceCount"`
```

Comment:
So `minDeviceCount` is not an optional field? It seems weird that user *must* specify a minimum number of disks

Reply:
we can keep it optional, then we should keep the default value to 1 right?

Reply:
cool